_client/src/Network/ApiManager.hpp

```diff
@@ -46,6 +51,9 @@ protected slots:
     void handleError(const QString &errorMessage);
     void handleLoginResponse(Method method, const Dataset &dataset);
     void handleRegistrationResponse(Method method, const Dataset &dataset);
+    void handleRoles(Method method, const Dataset &dataset);
+    void handleGetRolesResponse(Method method, const Dataset &dataset);
```

Comment:
You don't need this methods at all.
Just keep it `handleRoles` and do inside switch case to perform all necessary emits of signals